### PR TITLE
Fix enum array initializer using enum constant

### PR DESCRIPTION
**Pull request by zhangyiwei**
This is to fix https://github.com/KhronosGroup/Vulkan-Hpp/issues/902

### CODE REVIEW

**Comment by zhangyiwei** [comment]
@chrisforbes @oddhack @asuessenbach FYI, thanks!

**Comment by oddhack** [comment]
I don't know how Vulkan-Hpp is handling this case, but the purpose of the \<enum> tag in the array definition is to create a dependency forcing the array size to be defined prior to its use. This works in the C header generation framework, so perhaps this dependency isn't being handled by Vulkan-Hpp? Turning the token into its constant value in the header seems like a loss of useful information.

**Comment by zhangyiwei** [comment]
> I don't know how Vulkan-Hpp is handling this case, but the purpose of the <enum> tag in the array definition is to create a dependency forcing the array size to be defined prior to its use. This works in the C header generation framework, so perhaps this dependency isn't being handled by Vulkan-Hpp? Turning the token into its constant value in the header seems like a loss of useful information.

Initializer for enum array requires to explicitly construct the exact number of enum defaults. Previously, the dependency you mentioned is not needed in the Hpp generator because either the below covers everything.
1.  the array size is hardcoded to a number
2. those counting on the enum constant value doesn't necessarily require to generate such initializer but with an empty `{}`.

The case is: `std::array<VULKAN_HPP_NAMESPACE::ENUM_TYPE, ENUM_CONSTANT> const& some_instance_ = { { VULKAN_HPP_NAMESPACE::ENUM_TYPE::default, ...,  VULKAN_HPP_NAMESPACE::ENUM_TYPE::default} }`, which has exactly the number of `ENUM_CONSTANT` repeats inside the `{{ }}`.

**Review by asuessenbach — CHANGES_REQUESTED** [review]
Good approach changing the std::set of constants to a std::map.
But I would prefer to not pass that constant via appendEnumInitializer().

File: VulkanHppGenerator.cpp [hunk 1]
```diff
@@ -2505,7 +2506,7 @@ void VulkanHppGenerator::appendEnumInitializer( std::string &
   else
   {
     assert( arraySizes.size() == 1 );
-    int count = std::stoi( arraySizes[0] );
+    int count = std::stoi( usedConstant.empty() ? arraySizes[0] : m_constants.at(usedConstant) );
```
[the remark this answers is not in the document]
Comment (zhangyiwei) [reply VulkanHppGenerator.cpp]: Done

**Pull request merged**